### PR TITLE
Create sensible default settings for "Enable save actions" and "Cleanup" dialogs

### PR DESCRIPTION
**Pull request by motokito**
Sensible default settings for "Enable save actions" field of database properties dialog and "Run field formatter" field of Cleanup eintries dialog are now identical 🐨
Please take a look for merge. THX 🐰 

* Change the default setting - by click on "Reset" button:

2016-10-15_23h31_31

2016-10-15_23h31_45

* Change the default setting - by click on "Recommend for BibTeX" button:

2016-10-15_23h35_28

2016-10-15_23h36_12

* Change the default setting - by click on "Recommend for BibLaTeX" button:

2016-10-15_23h38_27

2016-10-15_23h38_40

For further information have a look at:
https://github.com/koppor/jabref/issues/128
😄 

- [x] Change in CHANGELOG.md described
- [x] Screenshots added (for bigger UI changes)
- [x] Manually tested changed features in running JabRef
- [x] Internal QS

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - Fixed [#1958](https://github.com/JabRef/jabref/issues/1958): Verbatim fields are no longer checked for HTML encoded characters by integrity checks
 - Fixed [#1937](https://github.com/JabRef/jabref/issues/1937): If no help page for the current chosen language exists, the english help page will be shown
 - Fixed [#1996](https://github.com/JabRef/jabref/issues/1996): Unnecessary other fields tab in entry editor removed (BibTeX mode)
+- Fixed [#128](https://github.com/koppor/jabref/issues/128): Sensible default settings for "Enable save actions" and "Cleanup"
```
Comment (koppor) [inline CHANGELOG.md]: Use `koppor/#128` as normal numbers reference to the JabRef /jabref repo.

Comment (motokito) [inline CHANGELOG.md]: I have done it 🐨

File: CHANGELOG.md [hunk 2]
```diff
@@ -81,7 +81,9 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - Fixed loop when pulling changes (shared database) when current selected field has changed
 - Fixed [#1958](https://github.com/JabRef/jabref/issues/1958): Verbatim fields are no longer checked for HTML encoded characters by integrity checks
 - Fixed [#1937](https://github.com/JabRef/jabref/issues/1937): If no help page for the current chosen language exists, the english help page will be shown
+- Fixed [#1996](https://github.com/JabRef/jabref/issues/1996): Unnecessary other fields tab in entry editor removed (BibTeX mode)
```
Comment (Siedlerchr) [inline CHANGELOG.md]: That line must be removed. Is probably from merge. However, the features was reverted

Comment (motokito) [inline CHANGELOG.md]: Done 😄

**Comment by koppor** [comment]
Sorry, but not all settings from the original dialog have been included:

grabbed_20160708-205241

- [ ] I miss the "convert to real superscripts". I think, this is "Ordinals to superscript"
- [ ] Please go through line by line and state which new converter you chose. Example:

 * `Run HTML converter on title` -> `title: HTML to LaTeX`

- [ ] I would also love to have a toggle button for "Unicode conversion", which adds/removes the respective formatters being equivalent to `Run Unicode converter on title, author(s), and abstract`.

- [ ] In case you are in context, could you check, why "Reformat ISSN" is there? Isn't that just another formatter?

Refs https://github.com/JabRef/help.jabref.org/issues/8

File: src/main/java/net/sf/jabref/logic/exporter/FieldFormatterCleanups.java [hunk 3]
```diff
@@ -42,8 +46,12 @@
 
         List<FieldFormatterCleanup> defaultFormatters = new ArrayList<>();
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.PAGES, new NormalizePagesFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.DATE, new NormalizeDateFormatter()));
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.MONTH, new NormalizeMonthFormatter()));
-        defaultFormatters.add(new FieldFormatterCleanup(FieldName.BOOKTITLE, new OrdinalsToSuperscriptFormatter()));
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/exporter/FieldFormatterCleanups.java]: Why was that removed?

**Comment by tobiasdiez** [comment]
I don't like all these "title" formatters. I would propose to have a minimal default preset which does not change to much. For example, "protect terms" is really hard to revert by hand. Thus I would only include the formatters which do a minimal job which probably everybody wants (i.e normalized months, dates and pages) but not more.

File: src/main/java/net/sf/jabref/logic/exporter/FieldFormatterCleanups.java (hunk 3 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/logic/exporter/FieldFormatterCleanups.java]: I have redone it 😄. That same i have insert in Cleanup entries dialog, when man click on reset button 😳 
But i have a question about the function of each checkbox. Which change will be setted, if i select one. Maybe like "Convert 1st, 2nd, ... to real superscript, what will be change if i select it and i approve with "OK". Thy for explain 😄

**Comment by motokito** [comment]
@koppor oliver what do you think about tobiasdiez idea ??? :smirk:

**Comment by motokito** [comment]
But i have a question about the function of each checkbox. Which change will be setted, if i select one. Maybe like "Convert 1st, 2nd, ... to real superscript, what will be change if i select it and i approve with "OK".
Thy for explaining 😄

**Comment by koppor** [comment]
To real superscripts is ordinals to latex. You could also have tried it out
by creating an inprocredings entry containing a bootitle such as "2nd
conference on something". Or just testing it with available bibtex files
and checking the deltas (e.g., using git)

**Comment by koppor** [comment]
@tobiasdiez We should have buttons for presets. Minimal, Recommended (as in
old JabRef versions) and Maximal (if we can come up w something useful)

I think, I really want the ordinals thing included. Don't like the protect
terms. Just want to put automatically braces around the complete title. But
we don't have a formatter for that...

**Comment by motokito** [comment]
@koppor Event without the checkbox "convert to real superscripts" you can add the entryfield manually to the fieldformatter with the converter you chose eg.

* Run Field Formatter -> choose entryfield -> booktitle: -> choose converter for superscripts -> for example HTML to Latex -> add

with this everyone can add his own converter for his own use thats why the checkbox "convert 1st, 2nd, ... to real superscripts" is not needed.

**Comment by koppor** [comment]
I know how to use the dialog. I want sensible defaults - see https://github.com/koppor/jabref/issues/128

File: CHANGELOG.md [hunk 4]
```diff
@@ -96,6 +96,8 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - Fixed [#2104](https://github.com/JabRef/jabref/issues/#2104): Crash after saving BibTeX source with parsing error
 - Fixed [#2109](https://github.com/JabRef/jabref/issues/#2109): <kbd>Ctrl-s</kbd> doesn't trigger parsing error message
 - Fixed RTFChars would only use "?" for characters with unicode over the value of 127, now it uses the base character (é -> e instead of ?)
+- Fixed [#1996](https://github.com/JabRef/jabref/issues/1996): Unnecessary other fields tab in entry editor removed (BibTeX mode)
```
Comment (boceckts) [inline CHANGELOG.md]: Remove this line.

Comment (motokito) [inline CHANGELOG.md]: Done 😄

**Comment by motokito** [comment]
The convert checkbox of old cleanup entries dialog is defined now in field formatter like following:
* Run HTML converter on title -> title: HTML to LateX
* Run Unicode converter on title, author(s), and abstract -> title: , author: or abstract: Unicode to LateX or LateX to Unicode
* Run filter on title keeping the case of selected words -> not found, what does this checkbox
* Remove unneccessary $, {, and } and move adjacent numbers into equations -> all_text_fields: LaTeX cleanup
* Add brackets and replace separators with their non-breaking version for units -> all_text_fields: Units to LaTeX
* Convert 1st, 2nd, ... to real superscripts -> all_text_fields: Ordinals to Latex superscript
* Format content of month field to #mon# -> month: Normalize month
* Ensure that page ranges are of the form num1--num2 -> pages: Normalize page numbers
* Format date field in the form yyyy-mm or yyyy-mm-dd -> date: Normalize date

All are done, please take a look for merge 🎅

**Comment by koppor** [comment]
Can the unicode converter (line 2) also be run on `all_text_fields`?

Run filter on title keeping the case --> This is "protect terms". Please enable it on title, journal, and journal title.

File: src/main/java/net/sf/jabref/model/entry/FieldName.java [hunk 5]
```diff
@@ -17,6 +17,9 @@
     // author is not set):
     public static final String FIELD_SEPARATOR = "/";
 
+    public static final String ABSTRACT_ALL_FIELD = "all";
+    public static final String ABSTRACT_ALL_TEXT_FIELDS_FIELD = "all_text_fields";
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/entry/FieldName.java]: Can this be named `all-text-fields`. I think dash ("`-`") is more intuitive than underscore ("`_`").

File: src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java [hunk 6]
```diff
@@ -73,6 +78,17 @@ public FieldFormatterCleanup(String field, Formatter formatter) {
         return fieldChanges;
     }
 
+    private List<FieldChange> cleanupAllTextFields(BibEntry entry) {
+        List<FieldChange> fieldChanges = new ArrayList<>();
+        Set<String> fields = entry.getFieldNames();
+        fields.removeAll(Arrays.asList(FieldName.DOI, FieldName.FILE, FieldName.URL, FieldName.URI));
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java]: Please introduce a class constant ALL_TEXT_FIELDS containing these fields. Please also add ISBN and ISSN.

**Comment by oscargus** [comment]
Some comments:

* Most bst:s doesn't change the case of Journal titles, so no need to protect them (please provide any counter examples if I am wrong).
* Adding braces around a complete field is in most cases the wrong way to do this as it will basically disable the bst handling of the field. (Note to @koppor )
* Using an all_text_fields meta field would be quite nice. I'd suggest HTML to LaTeX on those and maybe Unicode to LaTeX (although this is indeed really BibTeX/Biblatex dependent, therefore my hesitation, although we may have "recommended for BibTeX" and "recommended for Biblatex" versions...).

**Comment by tobiasdiez** [comment]
I'm really no big fan of the proposed default values:

- ProtectTermsFormatter: not needed in BibLaTex (and depends on the bibtex style)
- UnitsToLatexFormatter: I use the si-units package and thus this conversion is counterproductive.
- LatexCleanupFormatter: Is not good enough to be default in my opinion, but if the majority wants it then it would be fine with me
- HtmlToLatexFormatter: for BibLaTeX you want Html -> Unicode instead
- OrdinalsToSuperscriptFormatter: as a user of the (recommended nth-package)[http://tex.stackexchange.com/a/4119], this conversion is again suboptimal

In my opinion all these formatters depend on too much assumptions (BibTeX vs BibLaTeX and used packages) to be default. I find it better to have a minimal default preset and let the user decide which additional formatters he wants. Adding something feels better than removing something. 
But any way...its just default values.

File: src/main/java/net/sf/jabref/model/entry/FieldName.java (hunk 5 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/model/entry/FieldName.java]: Done :-)

File: src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java (hunk 6 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java]: Done 😄

**Comment by motokito** [comment]
@koppor: i have done all. Please take a look for merge. THX 🍪

**Comment by motokito** [comment]
@koppor: i have implemented recommend button for BibLatex /BibTex and refactoring the default setting
please take a look for merge BIG THX 😄

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java [hunk 7]
```diff
@@ -131,12 +135,32 @@ public void contentsChanged(ListDataEvent e) {
         resetButton = new JButton(Localization.lang("Reset"));
         resetButton.addActionListener(e -> ((CleanupActionsListModel) actionsList.getModel()).reset(defaultFormatters));
 
+        boolean isBibTex = !JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext().isBiblatexMode();
+        String mode;
+
+        if (isBibTex) {
+            mode = "BibTex";
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Must be "BibTeX"

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java [hunk 8]
```diff
@@ -131,12 +135,32 @@ public void contentsChanged(ListDataEvent e) {
...
+        } else {
+            mode = "BibLaTex";
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Must be "BibLaTeX"

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java [hunk 9]
```diff
@@ -131,12 +135,32 @@ public void contentsChanged(ListDataEvent e) {
...
+        }
+
+        recommendButton = new JButton(Localization.lang("Recommend for %0", mode));
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Should be "Recommended"

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 10]
```diff
@@ -18,17 +23,35 @@
...
 
         List<FieldFormatterCleanup> defaultFormatters = new ArrayList<>();
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.PAGES, new NormalizePagesFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.DATE, new NormalizeDateFormatter()));
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.MONTH, new NormalizeMonthFormatter()));
-        defaultFormatters.add(new FieldFormatterCleanup(FieldName.BOOKTITLE, new OrdinalsToSuperscriptFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.ABSTRACT_ALL_TEXT_FIELDS_FIELD, new OrdinalsToSuperscriptFormatter()));
         DEFAULT_SAVE_ACTIONS = new FieldFormatterCleanups(false, defaultFormatters);
+
+        List<FieldFormatterCleanup> recommendBibTexFormatters = new ArrayList<>(defaultFormatters);
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Also rename variable to `recommendedBibTeXFormatters`.

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 11]
```diff
@@ -18,17 +23,35 @@
...
+        recommendBibTexFormatters.add(new FieldFormatterCleanup(FieldName.TITLE, new HtmlToLatexFormatter()));
+        recommendBibTexFormatters.add(new FieldFormatterCleanup(FieldName.TITLE, new UnicodeToLatexFormatter()));
+        recommendBibTexFormatters.add(new FieldFormatterCleanup(FieldName.BOOKTITLE, new UnicodeToLatexFormatter()));
+        recommendBibTexFormatters.add(new FieldFormatterCleanup(FieldName.JOURNAL, new UnicodeToLatexFormatter()));
+        recommendBibTexFormatters.add(new FieldFormatterCleanup(FieldName.AUTHOR, new UnicodeToLatexFormatter()));
+        recommendBibTexFormatters.add(new FieldFormatterCleanup(FieldName.EDITOR, new UnicodeToLatexFormatter()));
+        RECOMMEND_BIBTEX_ACTIONS = new FieldFormatterCleanups(false, recommendBibTexFormatters);
+
+        List<FieldFormatterCleanup> recommendBibLatexFormatters = new ArrayList<>(defaultFormatters);
```
Comment (boceckts) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Also rename variable to `recommendedBibLaTeXFormatters`

Comment (motokito) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Done 😄

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java (hunk 9 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Done 😄

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java (hunk 8 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Done 😄

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java (hunk 7 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Done 😄

**Comment by koppor** [comment]
- [ ] "Recommended for bibtex" should be disabled if save actions are not enabled (similar to reset button)
- [ ] show screenshot for: "Reset"
- [ ] show screenshot for: "Recommended for bibtex"
- [ ] show screenshot for: "Recommended for biblatex"

**Comment by motokito** [comment]
@koppor: I have done it 😄:
* Change screenshots
* Disable "Recommend for %0" button, if save actions are not enable 🐨

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java [hunk 12]
```diff
@@ -193,8 +217,7 @@ private JPanel getSelectorPanel() {
                         "pref, 2dlu, pref:grow, 2dlu"));
 
         List<String> fieldNames = InternalBibtexFields.getAllPublicFieldNames();
-        fieldNames.add(BibEntry.KEY_FIELD);
-        fieldNames.add("all");
+        fieldNames.addAll(Arrays.asList(BibEntry.KEY_FIELD, FieldName.ABSTRACT_ALL_FIELD, FieldName.ABSTRACT_ALL_TEXT_FIELDS_FIELD));
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: I would do three calls using `fieldNames.add()`. This is IMHO more readable than `Arrays.asList`

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 13]
```diff
@@ -18,17 +23,35 @@
...
     static {
         availableFormatters = new ArrayList<>();
         availableFormatters.addAll(Formatters.ALL);
 
         List<FieldFormatterCleanup> defaultFormatters = new ArrayList<>();
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.PAGES, new NormalizePagesFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.DATE, new NormalizeDateFormatter()));
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.MONTH, new NormalizeMonthFormatter()));
-        defaultFormatters.add(new FieldFormatterCleanup(FieldName.BOOKTITLE, new OrdinalsToSuperscriptFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.ABSTRACT_ALL_TEXT_FIELDS_FIELD, new OrdinalsToSuperscriptFormatter()));
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Maybe we should stay with `BOOKTITLE` here.

File: src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java [hunk 14]
```diff
@@ -24,8 +27,10 @@ public FieldFormatterCleanup(String field, Formatter formatter) {
 
     @Override
     public List<FieldChange> cleanup(BibEntry entry) {
-        if ("all".equalsIgnoreCase(field)) {
+        if (FieldName.ABSTRACT_ALL_FIELD.equalsIgnoreCase(field)) {
             return cleanupAllFields(entry);
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java]: Is it possible to add test coverage here? - This branch is not covered by test cases.

File: src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java [hunk 15]
```diff
@@ -24,8 +27,10 @@ public FieldFormatterCleanup(String field, Formatter formatter) {
...
+        } else if (FieldName.ABSTRACT_ALL_TEXT_FIELDS_FIELD.equalsIgnoreCase(field)) {
+            return cleanupAllTextFields(entry);
```
Comment (koppor) [inline src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java]: please also cover by test cases

**Comment by koppor** [comment]
Please resolve conflicts in `JabRef_vi.properties` and please add test cases (see comments).

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java (hunk 12 above)
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: and extract it to a new method `InternalBibtexFields.getAllPublicAndInteralFieldNames()`.
Btw, I would suggest to rename the `ABSTRACT` prefix to `INTERNAL` (when reading "FieldName.abstract_Something" I always think of the "abstract" field)

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java (hunk 13 above)
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: I would completely remove the OrdinalsToSuperscript from the defaults.

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 16]
```diff
@@ -18,17 +23,35 @@
...
         DEFAULT_SAVE_ACTIONS = new FieldFormatterCleanups(false, defaultFormatters);
+
+        List<FieldFormatterCleanup> recommendedBibTeXFormatters = new ArrayList<>(defaultFormatters);
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.TITLE, new HtmlToLatexFormatter()));
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Is there a reason to not run the HTML/Unicode -> Latex converter on all [text] fields? Similar question for the Latex/HTML -> Unicode for biblatex.

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 17]
```diff
@@ -18,17 +23,35 @@
...
 
         List<FieldFormatterCleanup> defaultFormatters = new ArrayList<>();
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.PAGES, new NormalizePagesFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.DATE, new NormalizeDateFormatter()));
         defaultFormatters.add(new FieldFormatterCleanup(FieldName.MONTH, new NormalizeMonthFormatter()));
-        defaultFormatters.add(new FieldFormatterCleanup(FieldName.BOOKTITLE, new OrdinalsToSuperscriptFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.ABSTRACT_ALL_TEXT_FIELDS_FIELD, new OrdinalsToSuperscriptFormatter()));
         DEFAULT_SAVE_ACTIONS = new FieldFormatterCleanups(false, defaultFormatters);
+
+        List<FieldFormatterCleanup> recommendedBibTeXFormatters = new ArrayList<>(defaultFormatters);
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Make it more clear that default formatters are also added: 
recommendedbibTexFormatters = new ArrayList();
recommendendBibTexFormatters.addAll(defaulttFormatters);

(same for biblatex)

File: src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java [hunk 18]
```diff
@@ -73,6 +78,17 @@ public FieldFormatterCleanup(String field, Formatter formatter) {
         return fieldChanges;
     }
 
+    private List<FieldChange> cleanupAllTextFields(BibEntry entry) {
+        List<FieldChange> fieldChanges = new ArrayList<>();
+        Set<String> fields = entry.getFieldNames();
+        fields.removeAll(Arrays.asList(FieldName.DOI, FieldName.FILE, FieldName.URL, FieldName.URI, FieldName.ISBN, FieldName.ISSN, FieldName.MONTH, FieldName.DATE, FieldName.YEAR));
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/model/cleanup/FieldFormatterCleanup.java]: return these fields from a new method `FieldNames.getNotTextFieldNames()`

File: src/main/java/net/sf/jabref/preferences/JabRefPreferences.java [hunk 19]
```diff
@@ -1352,7 +1325,14 @@ public void setDefaultEncoding(Charset encoding) {
         put(DEFAULT_ENCODING, encoding.name());
     }
 
-    private static void insertCleanupPreset(Map<String, Object> storage, CleanupPreset preset) {
+    private static void insertCleanupPreset(Map<String, Object> storage) {
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/preferences/JabRefPreferences.java]: rename method to `insertDefaultCleanupPreset`

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java (hunk 16 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: @koppor what do you think about that ??? 😄

**Comment by koppor** [comment]
It is part of the old dialog, isn't it? It doesnt have to be part for the
minimum cleanups, but for the recommendations, it must!

**Comment by koppor** [comment]
Its my favourite. So keep it, please.

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 20]
```diff
@@ -46,12 +46,14 @@
         recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.JOURNAL, new UnicodeToLatexFormatter()));
         recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.AUTHOR, new UnicodeToLatexFormatter()));
         recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.EDITOR, new UnicodeToLatexFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.INTERNAL_ALL_TEXT_FIELDS_FIELD, new OrdinalsToSuperscriptFormatter()));
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: recommendedBibTeXFormatters instead of defaultFormatters (same below)

**Comment by motokito** [comment]
@koppor all thing are doing. Pls take a look for merge 🎅

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java [hunk 21]
```diff
@@ -18,17 +23,38 @@
...
+
+        List<FieldFormatterCleanup> recommendedBibTeXFormatters = new ArrayList<>();
+        recommendedBibTeXFormatters.addAll(defaultFormatters);
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.TITLE, new HtmlToLatexFormatter()));
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.TITLE, new UnicodeToLatexFormatter()));
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.BOOKTITLE, new UnicodeToLatexFormatter()));
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.JOURNAL, new UnicodeToLatexFormatter()));
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.AUTHOR, new UnicodeToLatexFormatter()));
+        recommendedBibTeXFormatters.add(new FieldFormatterCleanup(FieldName.EDITOR, new UnicodeToLatexFormatter()));
+        defaultFormatters.add(new FieldFormatterCleanup(FieldName.INTERNAL_ALL_TEXT_FIELDS_FIELD, new OrdinalsToSuperscriptFormatter()));
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: its still "defaultFormatters" here instead of "recommendedBibTeXFormatters"....

File: src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java [hunk 22]
```diff
@@ -131,12 +133,32 @@ public void contentsChanged(ListDataEvent e) {
         resetButton = new JButton(Localization.lang("Reset"));
         resetButton.addActionListener(e -> ((CleanupActionsListModel) actionsList.getModel()).reset(defaultFormatters));
 
+        boolean isBibTeX = !JabRefGUI.getMainFrame().getCurrentBasePanel().getDatabaseContext().isBiblatexMode();
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: work with "isBibLatex" instead of negating it.

Comment (motokito) [inline src/main/java/net/sf/jabref/gui/cleanup/FieldFormatterCleanupsPanel.java]: Done 😄

File: src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java (hunk 21 above)
Comment (motokito) [inline src/main/java/net/sf/jabref/logic/cleanup/Cleanups.java]: Done 😄

**Comment by koppor** [comment]
Good enough for me. Follow ups:
- [ ] remove empty lines before closing  `{` and if-statements
- [ ]  unifiy mode string generation with panel

Reason for second one: I assume that 
```
 if (isBibLaTeX) {
            mode = "BibLaTeX";
```
also appears somewhere else in the code.

grabbed_20161027-115448
